src/platform/forms-system/src/js/components/FormNav.jsx

```diff
@@ -11,82 +10,112 @@ import {
-        p => `${formConfig.urlPrefix}${p.path}` === currentPath,
-      );
+  let current;
+  let chapterName;
+  if (page) {
+    current = chapters.indexOf(page.chapterKey) + 1;
+    // The review page is always part of our forms, but isn’t listed in chapter list
+    chapterName =
+      page.chapterKey === 'review'
+        ? formConfig?.customText?.reviewPageTitle || REVIEW_APP_DEFAULT_MESSAGE
+        : formConfig.chapters[page.chapterKey].title;
+    if (typeof chapterName === 'function') {
+      chapterName = chapterName();
     }
+  }
 
-    let current;
-    let chapterName;
-    if (page) {
-      current = chapters.indexOf(page.chapterKey) + 1;
-      // The review page is always part of our forms, but isn’t listed in chapter list
-      chapterName =
-        page.chapterKey === 'review'
-          ? formConfig?.customText?.reviewPageTitle ||
-            REVIEW_APP_DEFAULT_MESSAGE
-          : formConfig.chapters[page.chapterKey].title;
-      if (typeof chapterName === 'function') {
-        chapterName = chapterName();
+  const stepText = `Step ${current} of ${chapters.length}: ${chapterName}`;
+  const showHeader = Math.abs(current - index) === 1;
+
+  // The goal with this is to quickly "remove" the header from the DOM, and
+  // immediately re-render the component with the header included.
+  // `current` changes when the form chapter changes, and when this happens
+  // we want to force react to remove the <h2> and re-render it. This should
+  // ensure that VoiceOver on iOS will pick up on the new <h2>
+  // https://github.com/department-of-veterans-affairs/va.gov-team/issues/12323
+  useEffect(
+    () => {
+      if (!environment.isProduction()) {
+        if (current > index + 1) {
+          setIndex(index + 1);
+        } else if (current === index) {
+          setIndex(index - 1);
+        }
       }
-    }
 
-    return (
-      <div>
-        <SegmentedProgressBar total={chapters.length} current={current} />
-        <div className="schemaform-chapter-progress">
-          <div
-            aria-valuenow={current}
-            aria-valuemin="1"
-            aria-valuetext={`Step ${current} of ${
-              chapters.length
-            }: ${chapterName}`}
-            aria-valuemax={chapters.length}
-            className="nav-header nav-header-schemaform"
-          >
-            {!environment.isProduction() ? (
-              <h2 className="vads-u-font-size--h4">
-                {`Step ${current} of ${chapters.length}: ${chapterName}`}
-              </h2>
-            ) : (
-              <h2 className="vads-u-font-size--h4">
-                <span className="form-process-step current">{current}</span>{' '}
-                <span className="form-process-total">of {chapters.length}</span>{' '}
-                {chapterName}
+      return () => {
+        if (!environment.isProduction()) {
+          focusElement('.nav-header > h2');
+        }
+      };
+    },
+    [current, index],
```

Comment:
These environment flags should prevent these effects from doing anything on production